dashboard/src/features/orgs/projects/services/components/ServiceForm/ServiceForm.tsx

```diff
@@ -88,14 +84,53 @@ export default function ServiceForm({
     watch,
     register,
     formState: { errors, isSubmitting, dirtyFields },
+    setValue,
   } = form;
 
   const formValues = watch();
 
-  const serviceImage = watch('image');
-
   const isDirty = Object.keys(dirtyFields).length > 0;
 
+  const newServiceID = useMemo(() => {
+    if (serviceID) {
+      return serviceID;
+    }
+    return uuidv4();
+  }, [serviceID]);
+
+  const privateRegistryImage = `registry.${project.region.name}.${project.region.domain}/${newServiceID}`;
+
+  let initialImageType: 'public' | 'private' | 'nhost' = 'public';
+
+  if (initialData?.image?.startsWith(privateRegistryImage)) {
+    initialImageType = 'nhost';
+  }
+
+  if (initialData?.pullCredentials?.length > 0) {
+    initialImageType = 'private';
+  }
+
+  const [imageType, setImageType] = useState<'public' | 'private' | 'nhost'>(
+    initialImageType,
+  );
+
+  const initialImageTag = useMemo(
+    () => initialData?.image?.split(':')[1],
+    [initialData?.image],
+  );
+
+  const handleImageTypeChange = (value: 'public' | 'private' | 'nhost') => {
+    if (value === initialImageType) {
+      setValue('image', initialData?.image);
+      setValue('pullCredentials', initialData?.pullCredentials);
+    } else {
+      setValue('image', '');
+      setValue('pullCredentials', undefined);
+    }
```

Comment:
minor thing, I think it's better to use `reset({image: ..., pullCredentials: ....})` to update multiple values at once